src/neo/SmartContract/BinarySerializer.cs

```diff
@@ -132,16 +132,16 @@ private static StackItem Deserialize(BinaryReader reader, uint maxArraySize, uin
             return stack_temp.Peek();
         }
 
-        public static byte[] Serialize(StackItem item, uint maxSize)
+        public static byte[] Serialize(StackItem item, uint maxSize, uint maxArraySize, uint maxItemSize)
         {
             using MemoryStream ms = new MemoryStream();
             using BinaryWriter writer = new BinaryWriter(ms);
-            Serialize(item, writer, maxSize);
+            Serialize(item, writer, maxSize, maxArraySize, maxItemSize);
             writer.Flush();
             return ms.ToArray();
         }
 
-        private static void Serialize(StackItem item, BinaryWriter writer, uint maxSize)
+        private static void Serialize(StackItem item, BinaryWriter writer, uint maxSize, uint maxArraySize, uint maxItemSize)
```

Comment:
Why add `maxArraySize` and `maxItemSize`? If the item exceeds the limit, it cannot be pushed onto the stack.

Reply:
This method it's shared with https://github.com/neo-project/neo/pull/1978/files#diff-03213bc090127a83f7aa1132e8e516e0R25 and it will be managed by the native contract not by neo-vm, so we can have different limmits

Reply:
Then we should limit in native contracts.

Reply:
Please, take a look to https://github.com/neo-project/neo/pull/1978